pubsub/google/cloud/pubsub_v1/publisher/batch/mp.py

```diff
@@ -0,0 +1,398 @@
+
+        Returns:
+            ~.pubsub_v1.types.Batching: The settings for batch
+                publishing. These should be considered immutable once the batch
+                has been opened.
+        """
+        return self._shared.settings
+
+    @property
+    def status(self):
+        """Return the status of this batch.
+
+        Returns:
+            str: The status of this batch. All statuses are human-readable,
+                all-lowercase strings.
+        """
+        return self._shared.status
+
+    def commit(self):
+        """Asychronously publish all of the messages on the active branch.
+
+        This method may be safely called from the primary process.
+        """
+        process = multiprocessing.Process(self._commit)
+        process.start()
+
+    def _commit(self):
+        """Actually publish all of the messages on the active batch.
+
+        This moves the batch out from being the active batch to an in-flight
+        batch on the publisher, and then the batch is discarded upon
+        completion.
+        """
+        # Update the status.
+        self._shared.status = 'in-flight'
+
+        # Begin the request to publish these messages.
+        if len(self._shared.messages) == 0:
+            raise Exception('Empty queue')
+        response = self._client.api.publish(
+            self._shared.topic,
+            self._shared.messages,
+        )
+
+        # FIXME (lukesneeringer): Check for failures; retry.
+
+        # We got a response from Pub/Sub; denote that we are processing.
+        self._status = 'processing results'
+
+        # Sanity check: If the number of message IDs is not equal to the
+        # number of futures I have, then something went wrong.
+        if len(response.message_ids) != len(self._shared.futures):
+            raise exceptions.PublishError(
+                'Some messages were not successfully published.',
+            )
+
+        # Iterate over the futures on the queue and return the response IDs.
+        # We are trusting that there is a 1:1 mapping, and raise an exception
+        # if not.
+        self._shared.status = self.Status.SUCCESS
+        for message_id, fut in zip(response.message_ids, self._shared.futures):
+            self._shared.message_ids[hash(fut)] = message_id
+            fut._trigger()
+
+    def monitor(self):
+        """Commit this batch after sufficient time has elapsed.
+
+        This simply sleeps for ``self._settings.max_latency`` seconds,
+        and then calls commit unless the batch has already been committed.
+        """
+        # Note: This thread blocks; it is up to the calling code to call it
+        # in a separate thread.
+        #
+        # Sleep for however long we should be waiting.
+        time.sleep(self.settings.max_latency)
+
+        # If, in the intervening period, the batch started to be committed,
+        # then no-op at this point.
+        if self.status != self.Status.ACCEPTING_MESSAGES:
```

Comment:
I feel like the current structure, with comments, follows my narrative logic better. But this is just a stylistic distinction.